### PR TITLE
Add encryption api endpoint to encrypt values

### PR DESCRIPTION
**Pull request by Suyash-d**
**ID**
187d769

Currently, we can encrypt data by providing the values to be encrypted through the terminal. 
It is better now to be able to execute this process from an API call through an endpoint.

### CODE REVIEW

File: core/java/iesi-rest-without-microservices/src/main/java/io/metadew/iesi/server/rest/encrypter/EncryptionController.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import javax.validation.Valid;
+
+/**
+ * The encryption controller is used to provide encrypted text.
+ * @author Suyash.d.jain
+ */
+
+@RestController
+@Tag(name = "encrypt", description = "controller to encrypt text")
+@RequestMapping("/encrypt")
```
Comment (HamzaaKhattabi) [inline core/java/iesi-rest-without-microservices/src/main/java/io/metadew/iesi/server/rest/encrypter/EncryptionController.java:21]: root endpoint should be a subject and in plural like "encryptions". Then, the post mapping should be the action such as "/encrypt". This way, we encrypt values by calling : /encryptions/encrypt. In the future we could have multiple services, so that controller should be generic.

We do the same with the login. Instead of creating a controller /login we do /users/login

File: core/java/iesi-rest-without-microservices/src/main/java/io/metadew/iesi/server/rest/encrypter/EncryptionController.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+public class EncryptionController {
+
+    private final IEncryptionService encryptionService;
+
+    public EncryptionController(IEncryptionService encryptionService) {
+        this.encryptionService = encryptionService;
+    }
+
+    @PostMapping("")
+    public EncryptDto getEncryptedPassword(@Valid @RequestBody EncryptPostDto encryptPostDto){
```
Comment (HamzaaKhattabi) [inline core/java/iesi-rest-without-microservices/src/main/java/io/metadew/iesi/server/rest/encrypter/EncryptionController.java:32]: Please call the method: encryptPassword. We don't really get the encrypted password, we encrypt the password and want the api to return that encrypted password

File: core/java/iesi-rest-without-microservices/src/main/java/io/metadew/iesi/server/rest/encrypter/IEncryptionService.java [hunk 3]
```diff
@@ -0,0 +1,6 @@
+package io.metadew.iesi.server.rest.encrypter;
+
+public interface IEncryptionService {
+
+    String getEncryptedPassword(String password);
```
Comment (HamzaaKhattabi) [inline core/java/iesi-rest-without-microservices/src/main/java/io/metadew/iesi/server/rest/encrypter/IEncryptionService.java:5]: please call it "encryptPassword"

File: core/java/iesi-rest-without-microservices/src/test/java/io/metadew/iesi/server/rest/encrypter/EncryptionControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+    @MockBean
+    private EncryptionService encryptionService;
+
+    @MockBean
+    private EncryptionController controller;
+
+    @Test
+    void getEncryptedPassword() {
+        controller.getEncryptedPassword(new EncryptPostDto("password"));
```
Comment (HamzaaKhattabi) [inline core/java/iesi-rest-without-microservices/src/test/java/io/metadew/iesi/server/rest/encrypter/EncryptionControllerTest.java:33]: Please test in detail that controller with the method `mvc.perform`.
Also test cases when providing empty text, etc.

**Review by HamzaaKhattabi — CHANGES_REQUESTED** [review]
Please rename `encrypter` to `encryption`

File: core/java/iesi-rest-without-microservices/src/test/java/io/metadew/iesi/server/rest/encrypter/EncryptionControllerTest.java (hunk 4 above)
Comment (Suyash-d) [reply core/java/iesi-rest-without-microservices/src/test/java/io/metadew/iesi/server/rest/encrypter/EncryptionControllerTest.java:33]: For Get method we write test case with mvc.perform for post we directly invoke. Have a look on existing test classes.

File: core/java/iesi-rest-without-microservices/src/main/java/io/metadew/iesi/server/rest/encrypter/EncryptionController.java (hunk 1 above)
Comment (Suyash-d) [reply core/java/iesi-rest-without-microservices/src/main/java/io/metadew/iesi/server/rest/encrypter/EncryptionController.java:21]: Login controller is different and Encrypt is different. and For Encryption we are not  going to create and update operation. So This will remain as it is. Have a look on Script and Connection controller, we don't write scripts/create, just /scripts and /connection to post new entity

File: core/java/iesi-rest-without-microservices/src/main/java/io/metadew/iesi/server/rest/encrypter/EncryptionController.java (hunk 2 above)
Comment (Suyash-d) [reply core/java/iesi-rest-without-microservices/src/main/java/io/metadew/iesi/server/rest/encrypter/EncryptionController.java:32]: We are passing some input and on the basic of that we are getting encrypted password. So name will remain getEncryptedPassword.

File: core/java/iesi-rest-without-microservices/src/test/java/io/metadew/iesi/server/rest/encrypter/EncryptionControllerTest.java (hunk 4 above)
Comment (HamzaaKhattabi) [reply core/java/iesi-rest-without-microservices/src/test/java/io/metadew/iesi/server/rest/encrypter/EncryptionControllerTest.java:33]: Yes, but how can you be sure if the encrypted data is encrypted properly ? we need to check that.

File: core/java/iesi-rest-without-microservices/src/main/java/io/metadew/iesi/server/rest/encrypter/EncryptionController.java (hunk 2 above)
Comment (HamzaaKhattabi) [reply core/java/iesi-rest-without-microservices/src/main/java/io/metadew/iesi/server/rest/encrypter/EncryptionController.java:32]: Yes but the action is to encrypt password not to get the encrypted password. When we create a script, he also have a scriptDto that has returned, and we don't call this method getCreatedScript.

File: core/java/iesi-rest-without-microservices/src/main/java/io/metadew/iesi/server/rest/encrypter/EncryptionController.java (hunk 1 above)
Comment (HamzaaKhattabi) [reply core/java/iesi-rest-without-microservices/src/main/java/io/metadew/iesi/server/rest/encrypter/EncryptionController.java:21]: Ok I agree with you first statement. But we could have in the future a method to get a text from an encrypted value. So the root endpoint should be /encryptions and not /encrypt. 

This way for : 
- Encrypting data we do : POST /encryptions/encrypt
- Decrypting data we do : POST /encryptions/decrypt

If we only create a controller named "encrypt", we assume that the controller ONLY encrypt data and nothing other.

File: core/java/iesi-rest-without-microservices/src/test/java/io/metadew/iesi/server/rest/encrypter/EncryptionControllerTest.java (hunk 4 above)
Comment (Suyash-d) [reply core/java/iesi-rest-without-microservices/src/test/java/io/metadew/iesi/server/rest/encrypter/EncryptionControllerTest.java:33]: We cant exactly check here because this post method, Please have a look on existing post method test cases.

Comment (HamzaaKhattabi) [reply core/java/iesi-rest-without-microservices/src/test/java/io/metadew/iesi/server/rest/encrypter/EncryptionControllerTest.java:33]: I had a look at the openApiController test, and tests case check the body response after performing post request as you can see below:
```
    @Test
    void transformFromBody() throws Exception {
        String jsonContent = new String(jsonFile);

        given(openAPIService.transform((MultipartFile) any())).willReturn(getTransformResult());
        given(transformResultDtoResourceAssembler.toModel(any())).willReturn(getTransformResultDto());

        mvc.perform(
                post("/openapi/transform")
                        .contentType(MediaType.APPLICATION_JSON)
                        .accept(MediaType.APPLICATION_JSON)
                        .content(jsonContent)
        )
                .andExpect(jsonPath("$").exists())
                .andExpect(jsonPath("$.title").isString())
                .andExpect(jsonPath("$.version").isString())
                .andExpect(jsonPath("$.connections").isArray())
                .andExpect(jsonPath("$.components").isArray())
                .andExpect(jsonPath("$.connections[0]").isMap())
                .andExpect(jsonPath("$.components[0]").isMap());
    }
```

**Review by HamzaaKhattabi — APPROVED** [review]
